managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3245,14 +3243,19 @@ public void setPropertiesFailed(ManagedLedgerException exception, Object ctx) {
     }
 
     @Override
-    public void asyncSetProperties(Map<String, String> properties, final SetPropertiesCallback callback, Object ctx) {
+    public void asyncSetProperties(Map<String, String> properties, boolean isOverwrite,
+                                   final SetPropertiesCallback callback,
+                                   Object ctx) {
+        propertiesMutex.lock();
```

Comment:
The properties feature in ManagedLedger is very broken as described in https://github.com/apache/pulsar/pull/7054#issuecomment-637843729. 

This PR is not addressing the overall issue, so I think it's better to hold off until a proper solution is in place.

Reply:
OK.